### PR TITLE
Issue 907 DatePicker states

### PR DESCRIPTION
**Pull request by daniel-martic-sociomantic**
For #907:

- added functionality and states to `DatePicker`
- removed `DatePicker` functionality from `DateTimeInput`

### CODE REVIEW

File: src/DatePicker/index.jsx [hunk 1]
```diff
@@ -7,70 +7,379 @@
...
+const DAY_LABELS = _.range( 0, 7 ).map( day => ( {
+    label : copy.dayHeaders[ day ],
+} ) );
+
+/**
+ * returns the timestamp of the current moment
+ *
+ * @return {Number} timestamp
+ */
+function now()
```
Comment (conor-cafferkey-sociomantic) [inline src/DatePicker/index.jsx]: Hmm not sure we need this function 🤔 Is there a difference between `new Date().getTime()` and `Date.now()`? If not, we could just use the latter directly

File: src/DatePicker/index.jsx [hunk 2]
```diff
@@ -7,70 +7,379 @@
...
+        const { max } = props;
+        const min = props.min || now();
+
+        if ( itemTimestamp > max ) return false;
+
+        return $m( itemTimestamp ).add( 1, unit ) > min;
+    };
+
+
+    const dayMatrix = () =>
```
Comment (conor-cafferkey-sociomantic) [inline src/DatePicker/index.jsx]: I don’t think this should be a function. I think we can just do:
```
const dayMatrix = _.chunk( days, 7 );
```
at the bottom

File: src/DatePicker/index.jsx [hunk 3]
```diff
@@ -7,70 +7,379 @@
...
+                isCurrent,
+                isDisabled,
+                isSelected,
+            };
+        } );
+
+        return _.chunk( days, 7 );
+    };
+
+    const monthMatrix = () =>
```
Comment (conor-cafferkey-sociomantic) [inline src/DatePicker/index.jsx]: Same as `dayMatrix`, I don’t think we need to define a function here.

File: src/DatePicker/index.jsx [hunk 4]
```diff
@@ -7,70 +7,379 @@
...
+            };
+        } );
+
+        return _.chunk( months, 4 );
+    };
+
+
+    const monthLabel = copy.months[ $m( gridStartTimestamp ).month() ];
+
+    const yearLabel = () => $m( gridStartTimestamp ).year().toString();
```
Comment (conor-cafferkey-sociomantic) [inline src/DatePicker/index.jsx]: This should just be: 
```
const yearLabel = $m( gridStartTimestamp ).year().toString();
```

(I don’t see any reason to wrap it in a function)

File: src/DatePicker/index.jsx (hunk 1 above)
Comment (daniel-martic-sociomantic) [reply src/DatePicker/index.jsx]: Indeed, `Date.now()` is a bit [better](https://stackoverflow.com/questions/12517359/performance-date-now-vs-date-gettime?answertab=votes#tab-top) and there's no need to have `function now()`

File: src/DatePicker/index.jsx (hunk 4 above)
Comment (daniel-martic-sociomantic) [reply src/DatePicker/index.jsx]: 👍

**Pull request merged**